libctru/include/3ds/applets/miiselector.h

```diff
@@ -0,0 +1,111 @@
+	                                  ///< Miis on launch.
+	char _unk0x8D[3];                 ///< @private
+	u32 initial_index;                ///< Index of the initially selected Mii. If
+	                                  ///< @ref MiiSelectorConf.show_guest_page is
+	                                  ///< set, this is the index of a Guest Mii,
+	                                  ///< otherwise that of a user Mii.
+	char mii_guest_whitelist[MIISELECTOR_GUESTMII_SLOTS]; ///< Each byte set to a nonzero value
+	                                                      ///< enables its corresponding Guest
+	                                                      ///< Mii to be enabled for selection.
+	char mii_whitelist[MIISELECTOR_USERMII_SLOTS]; ///< Each byte set to a nonzero value enables
+	                                               ///< its corresponding user Mii to be enabled
+	                                               ///< for selection.
+	u16 _unk0xFE;                                  ///< @private
+	u32 magic; ///< Will be set to @ref MIISELECTOR_MAGIC before launching the
+	           ///< applet.
+} MiiSelectorConf;
+
+/// @private
+_assert_struct_size(MiiSelectorConf, 0x104);
+
+/// Size of the data describing a single Mii
+#define MIISELECTOR_MIIDATA_SIZE 0x5c
+
+/// Maximum length of the localized name of a Guest Mii
+#define MIISELECTOR_GUESTMII_NAME_LEN 12
+
+/// Structure written by AppletEd
+typedef struct
+{
+	u32 no_mii_selected;                ///< 0 if a Mii was selected, 1 if the selection was
+	                                    ///< canceled.
+	u32 guest_mii_was_selected;         ///< 1 if a Guest Mii was selected, 0 otherwise.
+	u32 guest_mii_index;                ///< Index of the selected Guest Mii,
+	                                    ///< 0xFFFFFFFF if no guest was selected.
+	char mii[MIISELECTOR_MIIDATA_SIZE]; ///< Data of selected Mii.
+	u16 _pad0x68;                       ///< @private
+	u16 checksum;                       ///< Checksum of the returned Mii data.
+	                                    ///< Stored as a big-endian value; use
+	                                    ///< @ref miiSelectorChecksumIsValid to
+	                                    ///< verify.
+	u16 guest_mii_name[MIISELECTOR_GUESTMII_NAME_LEN]; ///< Localized name of a Guest Mii,
+	                                                   ///< if one was selected (UTF16-LE
+	                                                   ///< string). Zeroed otherwise.
+} MiiSelectorReturn;
+
+/// @private
+_assert_struct_size(MiiSelectorReturn, 0x84);
+
+#undef _assert_struct_size
+
+/**
+ * @brief Mii selector context
+ *
+ * Configure how to launch the applet in @ref MiiSelectorContext.config. After
+ * returning, the applet has written data to @ref MiiSelectorContext.ret.
+ */
+typedef union {
+	MiiSelectorConf config; ///< Configuration passed to applet
+	MiiSelectorReturn ret;  ///< Data written by applet after return
+} MiiSelectorContext;
```

Comment:
Instead of using a union, I would have miiSelectorLaunch() take two parameters, one for the input MiiSelectorConf and one for the output MiiSelectorReturn.